src/server/oicp/oicp-services-impl/oicp-2.3.0/OICPMapping.ts

```diff
@@ -0,0 +1,489 @@
+    evse.Accessibility = accessible;
+    evse.AccessibilityLocation; // Optional
+    evse.HotlinePhoneNumber = '+49123123123123'; // No information found for mandatory field
+    evse.AdditionalInfo; // Optional
+    evse.ChargingStationLocationReference; // Optional
+    evse.GeoChargingPointEntrance; // Optional
+    evse.IsOpen24Hours = true; // No information found for mandatory field
+    evse.OpeningTimes; // Optional
+    evse.ClearinghouseID; // Optional
+    evse.IsHubjectCompatible = true;
+    evse.DynamicInfoAvailable = OICPDynamicInfoAvailable.auto;
+    // Check addChargeBoxID flag
+    if (options && options.addChargeBoxID) {
+      evse.chargeBoxId = chargingStation.id;
+    }
+    // Return evse
+    return evse;
+  }
+
+  /**
+   * Convert ChargingStation to Multiple EVSE Statuses
+   * @param {Tenant} tenant
+   * @param {*} chargingStation
+   * @return Array of OICP EVSE Statuses
+   */
+  static convertChargingStation2MultipleEvseStatuses(tenant: Tenant, chargingStation: ChargingStation, options: { countryID: string; partyID: string; addChargeBoxID?: boolean}): OICPEvseStatusRecord[] {
+    // Loop through connectors and send one evse per connector
+    const connectors = chargingStation.connectors.filter((connector) => connector !== null);
+    const evseStatuses = connectors.map((connector) => {
+      const evseStatus = OICPMapping.convertConnector2EvseStatus(tenant, chargingStation, connector, options);
+      return evseStatus;
+    });
+    // Return all EVSE Statuses
+    return evseStatuses;
+  }
+
+  /**
+   * Convert Connector to EVSE Status
+   * @param {Tenant} tenant
+   * @param {*} connector
+   * @return Array of OICP EVSE Statuses
+   */
+  static convertConnector2EvseStatus(tenant: Tenant, chargingStation: ChargingStation, connector: Connector, options: { countryID: string; partyID: string; addChargeBoxID?: boolean}): OICPEvseStatusRecord {
+    const evseID = OICPUtils.buildEvseID(options.countryID, options.partyID, chargingStation, connector);
+    const evseStatus: OICPEvseStatusRecord = {} as OICPEvseStatusRecord;
+    evseStatus.EvseID = evseID;
+    evseStatus.EvseStatus = OICPMapping.convertStatus2OICPEvseStatus(connector.status);
+    // Check addChargeBoxID flag
+    if (options && options.addChargeBoxID) {
+      evseStatus.chargeBoxId = chargingStation.id;
+    }
+    return evseStatus;
+  }
+
+  /**
+   * Get evses from SiteArea
+   * @param {Tenant} tenant
+   * @param {SiteArea} siteArea
+   * @param options
+   * @return Array of OICP EVSES
+   */
+  static async getEvsesFromSiteaArea(tenant: Tenant, siteArea: SiteArea, options: { countryID: string; partyID: string; addChargeBoxID?: boolean }): Promise<OICPEvseDataRecord[]> {
+    // Build evses array
+    const evses: OICPEvseDataRecord[] = [];
+    // Convert charging stations to evse(s)
+    for (const chargingStation of siteArea.chargingStations) {
+      if (chargingStation.issuer === true && chargingStation.public) {
```

Comment:
I updated the logic partly.
If I there are connectors in the charge points array I send these connectors to Hubject (like the OCPI implementation)
But I cannot convertChargingStation2UniqueEvse and pool  the charge points in one EVSE.

I cannot map the charge point to the OICP EVSE the same way as in OCPI.
I saw that that the OCPIEvse type supports multiple connectors.

The OICPEvseDataRecord type only supports one connector. OICP uses the lowest level -> the connector.
I think the logic in case of a charging spot that cannot charge in parallel should be to update the connector status of the other connectors to unavailable.
That is the only way for OICP to ensure that nobody can authorize at this charge point when a connector is in use.

So I would send the charge point connectors as separate EVSEs and ensure in our backend that there is no parallel charging possible.

Reply:
You do not have to do it. The charging station or the backend should not try to authorise a session if the other mutually exclusive connector is used. 
That code in OCPI is merely a hack (and an ugly one) to give the same UID and/or ID to mutually exclusive connectors. 
You just have to send the EVSE OICP data to Hubject, all of them mapping each existing connectors and do not care about their exclusive usage. That's not the job of roaming code, it's job is to give the exhaustive topology of available connectors and use that information to all operations with the roaming API. 
And that's the job of the session authorisation code to cope with mutually exclusive connectors and/or the charging station.

Reply:
Ok, perfect. :D